pynisher/limiters/limiter.py

```diff
@@ -2,14 +2,32 @@
 from __future__ import annotations
 
 from abc import ABC, abstractmethod
-from typing import Any, Callable
+from typing import Any, Callable, Type
 
 import platform
 import sys
 import traceback
 from multiprocessing.connection import Connection
 
-from pynisher.util import Monitor
+from pynisher.exceptions import (
+    CpuTimeoutException,
+    MemoryLimitException,
+    PynisherException,
+    WallTimeoutException,
+)
+from pynisher.util import Monitor, callstring, terminate_process
+from pynisher.win_errcodes import WIN_ERROR_COMMITMENT_LIMIT
+
+
+def is_err(err: Exception, err_type: str | Type[Exception]) -> bool:
+    """Return if err is of a given type"""
+    # Check name of class matches
+    a = isinstance(err_type, str) and type(err).__name__ == err_type
+
+    # Explictly check for just class, not subclass
```

Comment:
Originally it was because we wrapped all builtin exceptions. Seeing as many people inherit from builtin exceptions such as `MyError(ValueError)`, this would end up catching them too. My use case was that a user might be using another library, say to catch one kind of error and silently not know that they're catching an entire subtree of exceptions inheriting from them. I'm not pressed either way but I guess it should be noted in the README.md. Do you have a preference?

Reply:
No, I don't have a preference, but I understand the comment now. Could you maybe add this explanation to the code and document the current behavior?

Reply:
Will do